components/camel-sql/src/main/java/org/apache/camel/component/sql/SqlProducer.java

```diff
@@ -69,8 +69,8 @@ public SqlEndpoint getEndpoint() {
     }
 
     @Override
-    protected void doStart() throws Exception {
-        super.doStart();
+    protected void doInit() throws Exception {
+        super.doInit();
```

Comment:
This should be doStart as SqlHelper.resolveQuery can potentially load the SQL template from file system. See the SQL docs